src/logstatsexp.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    logmeanexp(A; dims=:)
+
+Computes `log.(mean(exp.(A); dims))`, in a numerically stable way.
+"""
+function logmeanexp(A::AbstractArray; dims=:)
+    R = logsumexp(A; dims)
+    N = length(A) ÷ length(R)
+    return R .- log(N)
```

Comment:
This will cause undesired promotions and allocations.

Reply:
It seems you adressed my comment about peomotions but not the allocations.

Reply:
@devmotion you might argue that unless there's a way to modify `logsumexp` to directly take `log(N)` into account, an in-place operation might create issues with AD systems which do not support it?